gpio/gpioreset.go

```diff
@@ -11,37 +11,56 @@ func waitGPIO() {
 	time.Sleep(100 * time.Millisecond)
 }
 
-func pinctrlSet(pin string, state string) {
+func pinctrlSet(pin string, state string) error {
 	cmd := exec.Command("pinctrl", "set", pin, state)
-	if err := cmd.Run(); err != nil {
-		fmt.Printf("Error setting GPIO %s to %s: %v\n", pin, state, err)
+	output, err := cmd.CombinedOutput()
+	if err == nil {
+		return nil
 	}
+
+	return fmt.Errorf("Error setting GPIO %s to %s: %v output: %v", pin, state, err, string(output))
 }
 
-func InitGateway(resetPin, powerPin *int) {
+func InitGateway(resetPin, powerPin *int) error {
 	rst := strconv.Itoa(*resetPin)
 	var pwr string
 	if powerPin != nil {
 		pwr = strconv.Itoa(*powerPin)
 	}
-	initGPIO(rst, pwr)
-	resetGPIO(rst)
+	err := initGPIO(rst, pwr)
+	if err != nil {
+		return err
+	}
+	return resetGPIO(rst)
```

Comment:
olivia has a branch adding error handling, mind waiting for that?

Reply:
ok

Reply:
The error handling in this file is in main already, also made a ticket to handle the errors from the C library better.